src/app/shared/components/forecast-card/forecastCard.controller.js

```diff
@@ -1,13 +1,30 @@
 export default class ForecastCardController {
-    constructor() {
-
+    constructor($scope, $rootElement, $mdDialog) {
+        this.$scope = $scope;
+        this.$rootElement = $rootElement;
+        this.$mdDialog = $mdDialog;
     }
 
     getRotation() {
         if (this.currentForecast) {
             return `rotate(${this.currentForecast.wind.deg}deg)`;
         }
     }
+
+    openDialog(day) {
+        const dialogScope = angular.extend(this.$scope.$new(true), {
```

Comment:
check if this scope is destroyed with the dialog, in negative case, destroy it when dialog is closed